lightning/src/ln/channelmanager.rs

```diff
@@ -2559,14 +2552,19 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 	/// Value parameters are provided via the last hop in route, see documentation for RouteHop
 	/// fields for more info.
 	///
-	/// Note that if the payment_hash already exists elsewhere (eg you're sending a duplicative
-	/// payment), we don't do anything to stop you! We always try to ensure that if the provided
-	/// next hop knows the preimage to payment_hash they can claim an additional amount as
-	/// specified in the last hop in the route! Thus, you should probably do your own
-	/// payment_preimage tracking (which you should already be doing as they represent "proof of
-	/// payment") and prevent double-sends yourself.
+	/// If a pending payment is currently in-flight with the same [`PaymentId`] provided, this
+	/// method will error with an [`APIError::RouteError`]. Note, however, that once a payment
```

Comment:
it is a fairly common practice to catch idempotency error either on retries or in general and ignore it/ treat it as success for clients. 
Do you think we need a different sub-error?

Reply:
"verify that the send was identical to a previous one and return the same error as the previous one,"
-> that is true idempotency yes but too complex to implement for us. 

Alternatively, 
* We could just return something like "APIError::IdempotencyKeyCheckFailure".
* With that, can assume that request is already in process and ignore it, this is useful in duplicate requests scenario and retries (as in architecture scenario above)

Reply:
Otherwise, whenever there is an idempotency key failure,
user is left guessing:
* whether i should retry or not.
* whether i should ignore current request or not.

Reply:
are we saying we will add this later ?

Without this, Idempotency feature is unusable.

We did prevent duplicate payment but we did not resolve the second request.

In case of processing msgs from sqs, even in single node usecase, if i don't ignore duplicate requests it's going to keep retrying those msgs and eventually they end up in failure queue. (and those are not failures)

Reply:
Cool 👍